cmd/tetra/explain/explain.go

```diff
@@ -0,0 +1,452 @@
+		if err == nil {
+			fmt.Fprintf(w, "%s\n", indent(string(exampleData), "     "))
+		}
+	}
+
+	// Print fields
+	if recursive {
+		printFieldsRecursive(w, current, 0)
+	} else {
+		printFields(w, current)
+	}
+
+	return nil
+}
+
+func printFields(w io.Writer, s *apiextensionsv1.JSONSchemaProps) {
+	// If array, print fields of items
+	if s.Type == "array" && s.Items != nil {
+		s = s.Items.Schema
+	}
+
+	if len(s.Properties) > 0 {
+		fmt.Fprintf(w, "FIELDS:\n")
+		keys := make([]string, 0, len(s.Properties))
+		for k := range s.Properties {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		requiredSet := make(map[string]bool)
+		for _, req := range s.Required {
+			requiredSet[req] = true
+		}
+
+		for _, k := range keys {
+			prop := s.Properties[k]
+			required := ""
+			if requiredSet[k] {
+				required = " -required-"
+			}
+			fmt.Fprintf(w, "   %s\t<%s>%s\n", k, getTypeString(&prop), required)
+			if prop.Description != "" {
+				fmt.Fprintf(w, "%s\n", indent(formatDescription(prop.Description), "     "))
+			}
+			fmt.Fprintf(w, "\n")
+		}
+	} else if s.AdditionalProperties != nil {
+		fmt.Fprintf(w, "FIELDS:\n")
+		fmt.Fprintf(w, "   <map[string]%s>\n", getTypeString(s.AdditionalProperties.Schema))
+		if s.AdditionalProperties.Schema.Description != "" {
+			fmt.Fprintf(w, "%s\n", indent(formatDescription(s.AdditionalProperties.Schema.Description), "     "))
+		}
+		fmt.Fprintf(w, "\n")
+	}
+}
+
+func printFieldsRecursive(w io.Writer, s *apiextensionsv1.JSONSchemaProps, depth int) {
+	if depth > 5 { // Prevent infinite recursion
+		return
+	}
+
+	// If array, print fields of items
+	if s.Type == "array" && s.Items != nil {
+		s = s.Items.Schema
+	}
+
+	if len(s.Properties) > 0 {
+		if depth == 0 {
+			fmt.Fprintf(w, "FIELDS:\n")
+		}
+
+		keys := make([]string, 0, len(s.Properties))
+		for k := range s.Properties {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		requiredSet := make(map[string]bool)
+		for _, req := range s.Required {
+			requiredSet[req] = true
+		}
+
+		for _, k := range keys {
+			prop := s.Properties[k]
+			required := ""
+			if requiredSet[k] {
+				required = " -required-"
+			}
+
+			indentStr := strings.Repeat("  ", depth)
+			fmt.Fprintf(w, "%s   %s\t<%s>%s\n", indentStr, k, getTypeString(&prop), required)
+
+			if prop.Description != "" {
+				baseIndent := indentStr + "     "
+				fmt.Fprintf(w, "%s\n", indent(formatDescription(prop.Description), baseIndent))
+			}
+
+			// Recursive call for nested objects
+			if prop.Type == "object" && len(prop.Properties) > 0 {
+				printFieldsRecursive(w, &prop, depth+1)
+			} else if prop.Type == "array" && prop.Items != nil && len(prop.Items.Schema.Properties) > 0 {
+				printFieldsRecursive(w, prop.Items.Schema, depth+1)
+			}
+
+			fmt.Fprintf(w, "\n")
+		}
+	}
+}
+
+func formatDescription(desc string) string {
+	if desc == "" {
+		return ""
+	}
+
+	// Return the full description for proper multi-line handling
+	return desc
+}
```

Comment:
I'm not sure to understand why this function is needed, to my understanding it does nothing?